src/svd.jl

```diff
@@ -32,6 +32,88 @@ function svdvals(A::StaticMatrix)
     similar_type(A, T2, Size(diagsize(A)))(sv)
 end
 
+# Implement direct call to BLAS functions that computes the SVD values for `SMatrix` and
+# `MMatrix` reducing allocations. In this case, we use `MMatrix` to call the library and
+# convert the result back to the input type. Since the former does not exit this scope, we
+# can reduce allocations.
+#
+# We are implementing here the following functions:
+#
+#   svdvals(A::SMatrix{M, N, Float64}) where {M, N}
+#   svdvals(A::SMatrix{M, N, Float32}) where {M, N}
+#   svdvals(A::MMatrix{M, N, Float64}) where {M, N}
+#   svdvals(A::MMatrix{M, N, Float32}) where {M, N}
+#
+for (gesdd, elty) in ((:dgesdd_, :Float64), (:sgesdd_, :Float32)),
+    (mtype, vtype) in ((SMatrix, SVector), (MMatrix, MVector))
+
+    @eval begin
+        function svdvals(A::$mtype{M, N, $elty}) where {M, N}
+            K = min(M, N)
+
+            # Convert the input to a `MMatrix` and allocate the required arrays.
+            Am    = MMatrix{M, N, $elty}(A)
+            Sm    = MVector{K, $elty}(undef)
+
+            # We compute the `lwork` (size of the work array) by obtaining the maximum value
+            # from the possibilities shown in:
+            #   https://docs.oracle.com/cd/E19422-01/819-3691/dgesdd.html
+            lwork = max(8N, 3N + max(M, 7N), 8M, 3M + max(N, 7M))
+            work  = MVector{lwork, $elty}(undef)
+            iwork = MVector{8min(M, N), BLAS.BlasInt}(undef)
+            info  = Ref(1)
+
+            ccall(
```

Comment:
Can I suggest you use the `@ccall` macro here instead?

Reply:
Sure! I thought `@ccall` was introduced in Julia 1.7, but it was in 1.5.